gazebo_ros/include/gazebo_ros/conversions/gazebo_msgs.hpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_ROS__CONVERSIONS__GAZEBO_MSGS_HPP_
+#define GAZEBO_ROS__CONVERSIONS__GAZEBO_MSGS_HPP_
+
+#include <gazebo/msgs/contacts.pb.h>
+#include <gazebo_msgs/msg/contact_state.hpp>
+#include <gazebo_msgs/msg/contacts_state.hpp>
+
+#include <sstream>
+
+#include "gazebo_ros/conversions/builtin_interfaces.hpp"
+#include "gazebo_ros/conversions/generic.hpp"
+#include "gazebo_ros/conversions/geometry_msgs.hpp"
+
+namespace gazebo_ros
+{
+/// Generic conversion from an Gazebo Contact message to another type.
+/// \param[in] in Input message;
+/// \return Conversion result
+/// \tparam T Output type
+template<class T>
+T Convert(const gazebo::msgs::Contacts &)
+{
+  T::ConversionNotImplemented;
+}
+
+/// \brief Specialized conversion from an Gazebo message to a ROS Contacts State.
+/// \param[in] in Input message;
+/// \return A ROS Contacts state message with the same data as the input message
+template<>
+gazebo_msgs::msg::ContactsState Convert(const gazebo::msgs::Contacts & in)
+{
+  gazebo_msgs::msg::ContactsState contact_state_msg;
+  contact_state_msg.header.stamp = Convert<builtin_interfaces::msg::Time>(in.time());
+
+  ignition::math::Quaterniond frame_rot;
+  ignition::math::Vector3d frame_pos;
+
+  int contacts_packet_size = in.contact_size();
+  for (int i = 0; i < contacts_packet_size; ++i) {
+    // For each collision contact
+    // Create a ContactState
+    gazebo_msgs::msg::ContactState state;
+    const gazebo::msgs::Contact & contact = in.contact(i);
+
+    state.collision1_name = contact.collision1();
+    state.collision2_name = contact.collision2();
+    std::ostringstream stream;
+    stream << "Debug:  i: (" << i + 1 << "/" << contacts_packet_size <<
+      ")     my_geom: [" << state.collision1_name <<
+      "]  other_geom: [" << state.collision2_name <<
+      "]        time: [" << contact.time().sec() << "." << contact.time().nsec() << "]\n";
```

Comment:
Interesting that `in.contact(i).time()` (printed here) is different from `in.time()` (put in the header):

```
header:                                                                                                  
  stamp:                                                                                                 
    sec: 47                                                                                              
    nanosec: 466000000                                                                                   
  frame_id: world                                                                                        
states:                                                                                                  
- info: 'Debug:  i: (1/1)     my_geom: [ball::link::collision]  other_geom: [ground_plane::link::collisio
n]        time: [47.455000000]   
```

I think each contact has its own specific time, but the ROS message doesn't capture that.

This is just an observation, no action needed.